apps/services/license-api/src/app/modules/license/test/license.service.spec.ts

```diff
@@ -245,6 +307,75 @@ describe('LicenseService', () => {
           valid: true,
         })
       })
+
+      it.skip(`should verify barcodeData as token for the ${licenseId} license`, async () => {
+        // Act
+        const code = randomUUID()
+        const data = createCahceData(licenseId)
+
+        // Create token
+        const token = await barcodeService.createToken({
+          v: '1',
+          c: code,
+        })
+
+        // Put data in cache
+        await barcodeService.setCache(code, data)
+
+        const result = await licenseService.verifyLicense({
+          barcodeData: token,
+        })
+
+        // Assert
+        // Only driver's license is able to get extra data from the token for now
+        if (
+          licenseId === LicenseId.FIREARM_LICENSE ||
+          licenseId === LicenseId.DISABILITY_LICENSE
+        ) {
+          expect(result).toMatchObject({
+            valid: false,
+          })
+        } else {
+          expect(result).toMatchObject({
+            valid: true,
+            passIdentity: {
+              nationalId: data.nationalId,
+              name: data.extraData?.name,
+            },
+          })
+        }
+      })
+
+      it(`should fail to verify barcodeData token because of token expire time for the ${licenseId} license`, async () => {
+        // Act
+        const code = randomUUID()
+        const data = createCahceData(licenseId)
+
+        // Create token
+        const token = await barcodeService.createToken(
+          {
+            v: '1',
+            c: code,
+          },
+          { expiresIn: 1 },
+        )
+
+        // Put data in cache
+        await barcodeService.setCache(code, data)
+
+        // Wait for token to expire, before verifying
+        setTimeout(async () => {
```

Comment:
This runs after the test is finished. Should use `await sleep` or ideally use [fake timers](https://jestjs.io/docs/jest-object#fake-timers).